common/components/WalletDecrypt/components/WalletButton.tsx

```diff
@@ -1,23 +1,33 @@
 import React from 'react';
 import classnames from 'classnames';
-import { translateRaw } from 'translations';
+import { translateRaw, TranslateType } from 'translations';
 import { NewTabLink, Tooltip } from 'components/ui';
 import './WalletButton.scss';
+import { connect } from 'react-redux';
+import { isSupportedWalletFormat, getNetworkConfig } from 'selectors/config';
+import { WalletName } from 'config';
+import { AppState } from 'reducers';
 
-interface Props {
-  name: React.ReactElement<string> | string;
-  description?: React.ReactElement<string> | string;
-  example?: React.ReactElement<string> | string;
-  icon?: string | null;
-  helpLink?: string;
-  walletType: string;
+interface OwnProps {
+  name: TranslateType;
+  description?: TranslateType;
+  example?: TranslateType;
+  icon?: string;
+  helpLink: string;
+  walletType: WalletName;
   isSecure?: boolean;
   isReadOnly?: boolean;
   isDisabled?: boolean;
   onClick(walletType: string): void;
 }
 
-export class WalletButton extends React.Component<Props, {}> {
+interface StateProps {
+  isFormatDisabled?: boolean;
```

Comment:
Do we need to distinguish this?  Could just do `isDisabled: ownProps.isDisabled || ...` in `mapStateToProps`. But from my comment above about reducing connected components, maybe we should be doing this in WalletDecrypt and just have that keep track of `network`.

Reply:
Agreed, I believe I mentioned somewhere this could use some refactoring.

Reply:
Addressed in https://github.com/MyEtherWallet/MyEtherWallet/pull/849/commits/43b5833c54d5a87abc686501b8478014c8455e86

Reply:
Fixed in 366871c